modules/extension/xsd/xsd-wfs/src/main/java/org/geotools/wfs/PropertyValueCollection.java

```diff
@@ -180,7 +177,8 @@ public Object next() {
                             descriptor.isNillable(),
                             descriptor.getDefaultValue());
 
-            if (next instanceof SimpleFeature) {
+            if (next instanceof SimpleFeature || !(value instanceof ComplexAttribute)) {
+                if (value instanceof Attribute) value = ((Attribute) value).getValue();
```

Comment:
Caution, ``value instanceof Attribute`` will yield true for almost all attributes, since it's one of the root classes of the hierarchy. If I understood correctly, for simple feature the value should already be a concrete value (string, date, number, ...) but you step up in an use case where the value is an attribute so you want to extract the actual value?